src/packet/header/fragment_header.rs

```diff
@@ -53,10 +53,10 @@ impl FragmentHeader {
     }
 }
 
-impl HeaderParser for FragmentHeader {
+impl HeaderWriter for FragmentHeader {
     type Output = NetworkResult<()>;
 
-    fn parse(&self, buffer: &mut Vec<u8>) -> <Self as HeaderParser>::Output {
+    fn parse(&self, buffer: &mut Vec<u8>) -> <Self as HeaderWriter>::Output {
```

Comment:
This return type should be `Self::Output` iirc

Reply:
Again, one of those attempts to minimize the code change. I had fixed this in my rebuild.

Reply:
👍 if we can fix this in this PR that would be nice if not I dont mind

Reply:
👍